tst/gputests/TestUtils.h

```diff
@@ -0,0 +1,152 @@
+            break;
+        case Classification:
+            inpFeatureIndex.push_back(s % dataParameters.inpFeatureDimensionality); // one activation per sample
+            inpTime.push_back(s);
+
+            outFeatureIndex.push_back(s % dataParameters.outFeatureDimensionality); // one activation per sample
+            outTime.push_back(s);
+
+            vSampleTestInput.push_back(inpFeatureIndex);
+            vSampleTestInputTime.push_back(inpTime);
+            vSampleTestOutput.push_back(outFeatureIndex);
+            vSampleTestOutputTime.push_back(outTime);
+            break;
+        case ClassificationAnalog:
+            inpFeatureIndex.push_back(s % dataParameters.inpFeatureDimensionality); // one activation per sample
+            inpFeatureValue.push_back((float) s);
+            inpTime.push_back(s);
+
+            for (int d = 0; d < dataParameters.outFeatureDimensionality; d++) {
+                outFeatureIndex.push_back(d);
+                outFeatureValue.push_back(((s + d) % 2) + 1);
+                outTime.push_back(s);
+            }
+            vSampleTestInput.push_back(inpFeatureIndex);
+            vSampleTestInputData.push_back(inpFeatureValue);
+            vSampleTestInputTime.push_back(inpTime);
+
+            vSampleTestOutput.push_back(outFeatureIndex);
+            vSampleTestOutputData.push_back(outFeatureValue);
+            vSampleTestOutputTime.push_back(outTime);
+
+            break;
+        default:
+            std::cout << "unsupported mode";
+            exit(2);
+        }
+    }
+
+    int minInpDate = std::numeric_limits<int>::max(), maxInpDate = std::numeric_limits<int>::min(),
+                    minOutDate = std::numeric_limits<int>::max(), maxOutDate = std::numeric_limits<int>::min();
+    const bool alignFeatureNumber = false;
+    writeNETCDF(path + std::string("test.nc"), vSamplesName, mFeatureNameToIndex, vSampleTestInput, vSampleTestInputTime,
+                    vSampleTestInputData, mFeatureNameToIndex, vSampleTestOutput, vSampleTestOutputTime,
+                    vSampleTestOutputData, minInpDate, maxInpDate, minOutDate, maxOutDate, alignFeatureNumber, 2);
+}
+
+inline bool validateNeuralNetwork(const size_t batch, const std::string& modelPath, const TestDataType testDataType, const DataParameters& dataParameters) {
+    std::cout << "start validation of " << modelPath << std::endl;
+
+    NNNetwork* pNetwork = NULL;
+    std::vector<NNDataSetBase*> vDataSet;
+    std::string dataName = string("test.nc");
+    const std::string dataPath = "../tst/wrapper/toy_data/";
+    generateTestData(dataPath, testDataType, dataParameters);
+    vDataSet = LoadNetCDF(dataPath + dataName);
+    pNetwork = LoadNeuralNetworkJSON(modelPath, batch, vDataSet);
+    pNetwork->LoadDataSets(vDataSet);
+    pNetwork->SetCheckpoint("check", 1);
+    pNetwork->SetTrainingMode(SGD);
+    bool valid = pNetwork->Validate();
+    if (valid) {
+        std::cout << "SUCCESFUL validation" << std::endl;
```

Comment:
When these functions are called from tests that use cppunit, anything written to std::cout is going to interfere with cppunit's own console output. My suggestion is to update these functions to take a reference/pointer to a std::ostream, and to use that instead of std::cout or std::cerr.